lab_2/main.py

```diff
@@ -0,0 +1,133 @@
+import requests
+import csv
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from collections import Counter
+
+
+class MovieData:
+    api_path = 'https://api.themoviedb.org/3'
+    genre_url = f'{api_path}/genre/movie/list?language=en'
+    data_url = api_path + '/discover/movie?include_adult=false&include_video=false&sort_by=popularity.desc&page={!!!!}'
+    headers = {
+        'accept': 'application/json',
+        'Authorization': 'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIzMTI3NGFmYTRlNTUyMjRjYzRlN2Q0NmNlMTNkOTZjOSIsInN1YiI6IjVkNmZhMWZmNzdjMDFmMDAxMDU5NzQ4OSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.lbpgyXlOXwrbY0mUmP-zQpNAMCw_h-oaudAJB6Cn5c8'
+    }
+
+    def __init__(self, numb_of_pages):
+        self.genres = None
+        self.pages = numb_of_pages
+        self.data = []
+
+    def fetch_all_data_in_pages(self):
+        for i in range(self.pages):
+            response = requests.get(self.data_url, params={'page': i + 1}, headers=self.headers)
+            self.data.extend(response.json()['results'])
+
+    def give_data(self):
+        return self.data
+
+    def give_data_with_indexes(self):
+        return self.data[3:19:4]
+
+    def find_popular_title(self):
+        max_popularity = max(self.data, key=lambda x: x['popularity'])
+        popular_title = max_popularity['original_title']
+        return popular_title
+
+    def find_film_with_keyword(self, keyword):
+        title_list = [j['original_title'] for j in self.data if keyword in j['overview']]
+        return title_list
+
+    def get_unique_collections(self):
+        response = requests.get(self.genre_url, headers=self.headers)
+        self.genres = response.json()['genres']
+        genre_dict = {i['id']: i['name'] for i in self.genres}
+
+        all_genres = set()
+        for i in self.data:
+            genre_names = [genre_dict[j] for j in i['genre_ids'] if j in genre_dict]
+            all_genres.update(genre_names)
+        return all_genres
+
+    def delete_films_with_genre(self, genre_name_for_delete):
+        genre_id_for_delete = next(i['id'] for i in self.genres if i['name'].lower() == genre_name_for_delete.lower())
+        self.data = [i for i in self.data if genre_id_for_delete not in i['genre_ids']]
+        return self.data
+
+    def find_most_popular_genre(self):
+        all_genres = [genre_id for i in self.data for genre_id in i['genre_ids']]
+        genre_names = {genre['id']: genre['name'] for genre in self.genres}
+        most_common_genres = Counter(all_genres).most_common()
+        popular_genres = [{'genre_name': genre_names[i], 'count': count} for i, count in most_common_genres]
```

Comment:
how many element **most_common_genres** is going to contain?))
is it worth to create comprehension?)